homeassistant/components/miele/services.yaml

```diff
@@ -0,0 +1,33 @@
+# Services descriptions for Miele integration
+
+set_program:
+  target:
+    device:
+      integration: "miele"
+    entity:
+      integration: "miele"
+  fields:
+    program_id:
+      required: true
+      selector:
+        number:
+          min: 0
+          max: 99999
+          mode: box
+      example: 24
+    duration:
+      selector:
+        number:
+          min: 0
+          max: 240
+          mode: box
+          unit_of_measurement: min
+      example: 120
+    temperature:
+      selector:
+        number:
+          min: 100
+          max: 300
+          mode: box
+          unit_of_measurement: °C
+      example: 200
```

Comment:
I think I would recommend to split this into 2 services, one specific for ovens, since this way we at least avoid adding more things in the future imo

Reply:
I am not sure that it would make things easier for the users. Duration and temperature are optional for ovens. If you don't specify them, defaults for the selected program are used in the appliance. In addition it is not documented which models that support these parameters. I suggest that we keep it as on common `set_program` for all devices and await comments from the users. There has not been any complaints regaring this service action in the user base for the custom component.

Reply:
The thing I am most worried about is the slippery slope, so now we only have an oven, but in the future maybe an area for a robot vacuum or lawn mower, which in the end could make this service a mess.

Does every oven support this? Or only certain ones?

Reply:
It is not documented. But I have good reasons to believe that support is not complete across the entire portfolio. There are ovens, microwave ovens, steam/oven combos all with limited support in the public API. Then there are coffee systems, dishwashers, etc, The suggested service supports all known (as of today) implementations of set_program in the public API.

Reply:
Current service action is what is needed to support the most "controllable" ovens. A mandatory program_id, and optional duration and temperature. I don't see the point with a specific service for ovens. The API has been stable in this area since the beginning.

Reply:
> but in the future maybe an area for a robot vacuum

Sending a robot vacuum cleaner to a specific area is handled by another API endpoint so I think it will be logical to have a separate service action for that purpose.